src/components/Calendar/Calendar.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import {themr} from 'react-css-themr';
+import moment from 'moment';
+import Month from './Month';
+import {PURE} from 'dx-util/src/react/react';
+import ButtonIcon from '../ButtonIcon/ButtonIcon';
+import {MEMOIZE} from 'dx-util/src/function/function';
+import {CALENDAR_THEME} from './Calendar.constants';
+import noop from '../../util/func/noop';
+
+export const CALENDAR = Symbol('Calendar');
+
+@PURE
+@themr(CALENDAR)
+export default class Calendar extends React.Component {
+	static propTypes = {
+		value: React.PropTypes.string.isRequired, // ISO - "2016-09-20T15:30:39.298Z"
+		headerDateFormat: React.PropTypes.string,
+		headerDayFormat: React.PropTypes.string,
+		dayFormat: React.PropTypes.string,
+		onChange: React.PropTypes.func,
+		min: React.PropTypes.string, // ISO
+		max: React.PropTypes.string, // ISO
+		previousMonthIcon: React.PropTypes.string,
+		nextMonthIcon: React.PropTypes.string,
+		locale: React.PropTypes.string,
+		theme: React.PropTypes.shape(CALENDAR_THEME)
+	}
+
+	static defaultProps = {
+		onChange: noop,
+		min: null,
+		max: null,
+		headerDateFormat: 'MMM YYYY',
```

Comment:
all formatter should be functions of type `type formatter = (date: Date)=> string`. then we can reduce dependence on moment lib